src/passwd.c

```diff
@@ -740,17 +736,7 @@ static int check_selinux_access (const char *changed_user,
 	if (changed_uid != 0 && strcmp(changed_user, user) == 0) {
```

Comment:
`if (changed_uid != 0 && strcmp(changed_user, user) == 0)`
Is this shortcut really necessary?
Also the comment above does not match.

Reply:
This logic appears to be a legacy of original SELinux reference implementation where SELinux user identity (i.e. first field of the security context) was the same as the Linux username.  Then the logic tries to require that in order to change the password of a different user, one needs to have the "passwd" permission.  Then someone further amended it to also require the "passwd" permission if changing the root password regardless. Seems like this will almost always fail since the SELinux user identity is rarely (never?) the same as the Linux username and thus we will always execute the else clause.  Likely should be removed.

Reply:
Legacy shortcut dropped